### ISSUE TITLE
double parentheses on template strings

### ISSUE THREAD

**Issue by manferlo81** [bublejs/buble#201]
I am getting a double parentheses when using template strings.

The following is just a dummy example...

```javascript
function logInvalid(obj) {
  console.log(`${obj} is not valid.`);
}
```
gets compile to...
```javascript
function logInvalid(obj) {
  console.log((obj + " is not valid."));
}
```
I am not sure that's the intended behavior.

see the life example [here](https://buble.surge.sh/#function%20logInvalid(obj)%20%7B%0A%20%20console.log(%60%24%7Bobj%7D%20is%20not%20valid.%60)%3B%0A%7D)

**Comment by nathancahill** [comment]
The added parentheses are to preserve the order of operations once the template string it compiled. The alternative is to do what Babel does: `console.log("".concat(obj, " is not valid."))`

Is this causing unexpected behavior?

**Comment by manferlo81** [comment]
@nathancahill, I dont think it has to do with order of operation because there is no order to preserve in a single operator expression. It would apply in the following case...

```javascript
console.log(`The result of the sum is ${5 + 2}.`);
```

I expect...

```javascript
console.log("The result of the sum is " + (5 + 2) + ".");
```

But I am getting...

```javascript
console.log(("The result of the sum is " + (5 + 2) + "."));
```

See [here](https://buble.surge.sh/#function%20logInvalid(obj)%20%7B%0A%20%C2%A0console.log(%60The%20result%20of%20the%20sum%20is%20%24%7B5%20%2B%202%7D.%60)%3B%0A%7D)

 Also Babel is giving me the result I expect...

```javascript
console.log(obj + " is not valid.");
// and
console.log("The result of the sum is " + (5 + 2) + ".");
```

Can you please explain on what situation this extra outer parentheses are useful?

**Comment by nathancahill** [comment]
Parentheses are needed. See this simple example:

ES6

```
> var a = 'a'
> var b = 'b'
> `${a}${b}`.length
< 2
```

Transpiled without parentheses: 

```
> a + b.length
< 'a1'
```

Transpiled with parentheses:

```
> (a + b).length
< 2

**Comment by mourner** [comment]
I guess this is more of a purity/redundancy issue — parentheses are needed when the template literal is a part of a more complex expression, but not e.g. when surrounded by parentheses already. But this really doesn't matter in the end because minifiers will strip this away.
